var/spack/repos/builtin/packages/py-gmxapi/package.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import os
+
+from spack.package import *
+
+
+class PyGmxapi(PythonPackage):
+    """Python bindings and ensemble workflow management for GROMACS.
+
+    The GROMACS C++ API is affected by its package variants. You can
+    specify a particular GROMACS API by making the dependency variant explicit.
+    E.g. ``spack install gmxapi ^gromacs+mpi~double``
+    """
+
+    homepage = "https://manual.gromacs.org/current/gmxapi/index.html"
+    maintainers("eirrgang", "peterkasson")
+
+    pypi = "gmxapi/gmxapi-0.4.0.tar.gz"
+    version("0.4.0", sha256="7fd58e6a4b1391043379e8ba55555ebeba255c5b394f5df9d676e6a5571d7eba")
+
+    depends_on("gromacs@2022.1:~mdrun_only+shared")
+    depends_on("mpi")
+    depends_on("py-cmake@3.16:", type="build")
+    depends_on("py-importlib-metadata", type="test", when="^python@:3.7")
+    depends_on("py-mpi4py", type=("build", "run"))
+    depends_on("py-networkx@2.0:", type=("build", "run"))
+    depends_on("py-numpy@1.7:", type=("build", "run"))
+    depends_on("py-setuptools@42:", type="build")
+    depends_on("py-packaging", type=("build", "run"))
+    depends_on("py-pybind11@2.6:", type="build")
```

Comment:
```suggestion
    depends_on("py-pybind11@2.6:", type=("build", "run"))
```
According to `setup.cfg`

Reply:
True, but setup.cfg is wrong and will be fixed in 0.4.1. Is there some sort of metadata checking that will cause a problem if this change is not made for 0.4.0?

Reply:
At the moment, nothing automatic. I've been thinking about adding the `--check-build-dependencies` flag to `pip install` which might complain, and running `pip check` as a post-install test, which will definitely complain, but haven't added those yet. I usually err on the side of following the documented deps, even if they are wrong. But it's up to you.

Reply:
`pip check` would break this if done in a different environment than the (non-isolated) build environment.

I vote for caution.